bundle-workflow/python/test_workflow/test_cluster.py

```diff
@@ -9,4 +31,21 @@ def port(self):
         return 9200
 
     def destroy(self):
-        pass
+        print(f'Sending SIGTERM to PID {self.process.pid}')
+        self.process.terminate()
+        try:
+            print('Waiting for process to terminate')
+            self.process.wait(10)
+        except TimeoutExpired:
+            print('Process did not terminate after 10 seconds. Sending SIGKILL')
+            self.process.kill()
+            try:
+                print('Waiting for process to terminate')
+                self.process.wait(10)
+            except TimeoutExpired:
+                print('Process failed to terminate even after SIGKILL')
+                raise
+        print(f'Process terminated with exit code {self.process.returncode}')
+        self.stdout.close()
```

Comment:
If the process times out on termination, you'll be leaking a handle.

Reply:
Good catch

Reply:
Fixed